main.rb

```diff
@@ -21,75 +21,82 @@ module BstOperation
 end
     
 def show_bst_operations
-    total_op = BST_OPERATION.length
-
-    (1..total_op).each do |op|
-        print op, ' => ', BST_OPERATION[op]
+    BST_OPERATION.each do |key, msg|
+        print key, ' => ', msg
         puts
     end
     print 'Please enter your choice : '
 end
 
-def run_bst_script(tree, user_input)
-
-    if user_input == 'quit'
-        user_input = 10
-    else
-        user_input = user_input.to_i
+def is_valid_input(elements)
+    begin
+        elements = elements.map { |elem| Integer(elem) }
+    rescue ArgumentError
+        return false
     end
+    true
+end
+
+def run_bst_script(tree, user_input)
+    
+    ops = [BstOperation::INSERT, BstOperation::LOAD,BstOperation::SAVE, BstOperation::QUIT]
+    return puts "Empty Tree" if tree.empty? and !ops.include?(user_input)
```

Comment:
```
if tree.empty? and !ops.include?(user_input)
    puts "Empty Tree"    
    return 
end
```

Reply:
is there any problem with the single liner here?